events/interactionCreate.js

```diff
@@ -1,15 +1,40 @@
 const logger = require("../modules/Logger.js");
+const { getSettings, permlevel } = require("../modules/functions.js");
+const config = require("../config.js");
+
 module.exports = async (client, interaction) => {
   // If it's not a command, stop.
   if (!interaction.isCommand()) return;
+
+  // Grab the settings for this server from Enmap.
+  // If there is no guild, get default conf (DMs)
+  const settings = interaction.settings = getSettings(interaction.guild);
+
+  // Get the user or member's permission level from the elevation
+  const level = permlevel(interaction);
+  
   // Grab the command data from the client.container.slashcmds Collection
   const cmd = client.container.slashcmds.get(interaction.commandName);
+  
   // If that command doesn't exist, silently exit and do nothing
   if (!cmd) return;
-  // Run the command
+
+  // Since the permission system from Discord is rather limited in regarding to
+  // Slash Commands, we'll just utilise our permission checker.
+  if (level < client.container.levelCache[cmd.conf.permLevel]) {
+    if (settings.systemNotice === "true") {
+      return await interaction.reply(`You do not have permission to use this command.
+Your permission level is ${level} (${config.permLevels.find(l => l.level === level).name})
+This command requires level ${client.container.levelCache[cmd.conf.permLevel]} (${cmd.conf.permLevel})`);
+    } else {
+      return;
```

Comment:
If systemNotice is false then instead of `retun;` can we return the response with `ephemeral: true`
```
else {
  return await interaction.reply({ content: `You do not have permission to use this command.
Your permission level is ${level} (${config.permLevels.find(l => l.level === level).name})
This command requires level ${client.container.levelCache[cmd.conf.permLevel]} (${cmd.conf.permLevel})`, ephemeral: true });
}
```

Reply:
if we wont reply then the interaction will fail

Reply:
Good point, I'll add that now